### PR TITLE
test(drawer): refactor for tests

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

Refactor for `Drawer` component to use new cypress-component-react framework for testing.

### Current behaviour
Currently we are using the regular cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- Run npx cypress open-ct to check if there is newly added test.file
 - Check if the `drawer.test.js` file passed
 - Compare all tests were moved/refactored from .feature file -> .test.js file
 - Check if all unnecessary files were deleted, including .json file

### CODE REVIEW

File: src/components/drawer/drawer.test.js [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+      }
+    );
+
+    it("should render component closed by default", () => {
+      CypressMountWithProviders(<DrawerCustom defaultExpanded={false} />);
+      drawerSidebarContentInnerElement(positionOfElement("first"))
+        .should("have.text", "link a")
+        .and("not.be.visible");
+    });
+
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply src/components/drawer/drawer.test.js:95]: there is no point to do that, because it is set to `true` be default

**Review by DianeGit22 — DISMISSED** [review]

File: src/components/drawer/drawer.test.js [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+      drawer().should("have.attr", "height", "75%");
+    });
+
+    it("should render component with custom title", () => {
+      CypressMountWithProviders(<DrawerCustom title="cypress_title" />);
+
+      drawer().children().children().eq(0).should("have.text", "cypress_title");
+    });
+
+    it("should render component with showControls", () => {
```
Comment (edleeks87) [inline src/components/drawer/drawer.test.js:155]:
```suggestion
    it("should render component with toggle control when showControls prop is true", () => {
```

File: src/components/drawer/drawer.test.js [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+      CypressMountWithProviders(<DrawerCustom showControls />);
+
+      drawerToggle().should("be.visible");
+    });
+
+    it("should render component without showControls prop", () => {
```
Comment (edleeks87) [inline src/components/drawer/drawer.test.js:161]:
```suggestion
    it("should render component without toggle control when showControls prop is falsy", () => {
```

File: src/components/drawer/drawer.test.js [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+    });
+
+    it("should render component with toggle control when showControls prop is true", () => {
+      CypressMountWithProviders(<DrawerCustom showControls />);
+
+      drawerToggle().should("be.visible");
+    });
+
+    it("should render component without toggle control when showControls prop is falsy", () => {
+      CypressMountWithProviders(<DrawerCustom />);
```
Comment (DianeGit22) [inline src/components/drawer/drawer.test.js:162]: just a typo here "falsy"

**Review by DianeGit22 — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**